### ISSUE TITLE
Unexpected default vendor accept header.

### ISSUE THREAD

**Issue by yoonsy** [EDMC/api-versions#25]
[According to a middleware spec](https://github.com/EDMC/api-versions/blob/f55068fc2fb6098ae1d09664b6dfc7a0a032d1ef/spec/middleware_spec.rb#L27), `ApiVersions::Middleware` should add a default vendor accept to a nil Accept header.

But, in my opinion, it adds default vendor only if request path contains prefix path like `/api`. Otherwise, It adds default vendor when Accept header is `*/*`.

What do you think of this?

Thank you for sharing this project. :+1:

**Comment by yoonsy** [comment]
I forked this project and changed the code [like this](https://github.com/yoonsy/api-versions/commit/8cffabb9e9a1587564b4e87b90bb7f2579cac827).

I fixed the api prefix to `/api`, so I didn't open PR.

**Comment by rposborne** [comment]
Agreed, adding the header to all can make controller tests suddenly think they are getting json. (when they have nothing to do with the API)
